### PR TITLE
feat(sponnet): camelCase, additional artifacts and stages

### PR DESCRIPTION
**Pull request by kskewes**
- Style: camelCase stage types and demo.jsonnet variables.
- Add additional artifact types.
- Add Kubernetes FindArtifactsFromResource stage
- fix Kubernetes DeployManifest stage correctly sets source artifact or text
  based on whether artifact supplied or not.
- Extend Jenkins stage - add properties file and configuration options
- Extend Wait stage - add custom skip message
- Add pipeline stage
- Add Wercker stage, untested as don't have access to a Wercker install.

Picked up some missing configuration options from initial stage add.

Moving on now to putting into use internally. Will come back to add more missing stages and options 
 in a future PR later.

### CODE REVIEW

**Comment by kskewes** [comment]
Forgot need to refactor notifications before we can put to use here. Almost done will send commit tomorrow.

**Comment by kskewes** [comment]
@lwander - Would you mind having a look when free?
Sorry the first commit message was ambiguous. Will be a few weeks/months+ before can look at additional stages and things.

File: sponnet/pipeline.libsonnet [hunk 1]
```diff
@@ -63,26 +69,23 @@
   },
 
   // notifications
-  // TODO: refactor level to include stage level.
 
-  notification(address, type, conditions, when, message):: {
+  notification:: {
     withAddress(address):: self + { address: address },
-    level: 'pipeline',
+    withCC(address):: self + { address: address },
```
Comment (lwander) [inline sponnet/pipeline.libsonnet]: I address used by all notification types, not just email?

Comment (kskewes) [reply sponnet/pipeline.libsonnet]: Yes. `address` field is used for hipchat room, slack room, sms phone number, email address, etc.
Only email has an additional field available to it (`cc`) and have just pushed the naming fix for that now.
I hadn't bothered to split out `notification` into different types but can do. Might be more useful when the Github stuff comes along.

**Pull request merged**